compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -223,16 +223,16 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
   /** Create an anonymous class `new Object { type MirroredMonoType = ... }`
    *  and mark it with given attachment so that it is made into a mirror at PostTyper.
    */
-  private def anonymousMirror(monoType: Type, attachment: Property.StickyKey[Unit], span: Span)(using Context) =
+  private def anonymousMirror(monoType: Type, mirroredType: Type, attachment: Property.StickyKey[Unit], pre: Type, span: Span)(using Context) =
     if ctx.isAfterTyper then ctx.compilationUnit.needsMirrorSupport = true
     val monoTypeDef = untpd.TypeDef(tpnme.MirroredMonoType, untpd.TypeTree(monoType))
-    val newImpl = untpd.Template(
+    var newImpl = untpd.Template(
       constr = untpd.emptyConstructor,
       parents = untpd.TypeTree(defn.ObjectType) :: Nil,
       derived = Nil,
       self = EmptyValDef,
       body = monoTypeDef :: Nil
-    ).withAttachment(attachment, ())
+    ).withAttachment(attachment, ()).withAttachment(AnonymousMirrorPrefix, pre)
```

Comment:
@bishabosha I want to make anonymous mirrors SAMs and I'm thinking - wouldn't it be easier to synthesize them here directly instead of passing all the attachments and then manually linking the symbols?

Reply:
I think it would be in principle, however there is supposed to be a feature that the companion object is not meant to leak the mirror definitions if its the mirror (its a bit broken now) - which would be difficult if mirror implementation was added in typer

Reply:
But that shouldn't be affected by synthesizing anonymous mirrors in the typer.

Reply:
yes I think that can work - I'm not aware of the original reason to do this

Reply:
^ reserve this work for another PR - probably involves abstracting a few things in `SyntheticMembers` (where anonymous mirrors get their definitions)

Reply:
Yes, I meant another PR - but I can't seem to find the time